src/metabase/data_app/api/data_app.clj

```diff
@@ -0,0 +1,108 @@
+(ns metabase.data-app.api.data-app
+  "/api/data-app endpoints for data apps CRUD operations"
+  (:require
+   [honey.sql.helpers :as sql.helpers]
+   [metabase.api.common :as api]
+   [metabase.api.macros :as api.macros]
+   [metabase.data-app.models :as data-app.models]
+   [metabase.request.core :as request]
+   [metabase.util :as u]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(defn- get-data-app
+  "Get a data app by id with latest definitions and release info."
+  [id]
+  (let [data-app (api/read-check (t2/select-one :model/DataApp id))]
+    (assoc data-app
+           :latest_definition (data-app.models/latest-definition id)
+           :released_definition (data-app.models/released-definition id)
+           :latest_release (data-app.models/latest-release id))))
```

Comment:
Let's rather return a release summary. I don't think we need to expose the snapshot ids or anything, I think we can just return:

```ts
[{id, restracted, published_at}]
```

Basically we'll want the user to be able to see what's released, and have an option to retract versions.

We could also YAGNI this for now.